src/rpc/test_serialize.zig

```diff
@@ -23,33 +23,46 @@ const GetVoteAccounts = methods.GetVoteAccounts;
 const Response = rpc.response.Response;
 
 fn testRequest(
+    comptime method: methods.MethodAndParams.Tag,
     /// passed into the client
-    request: anytype,
+    // TODO: use this instead of `std.meta.FieldType` to avoid eval branch quota until `@FieldType`'s here.
+    params: @typeInfo(methods.MethodAndParams).Union.fields[@intFromEnum(method)].type,
     /// test will assert the request serializes to this json
     expected_request_json: []const u8,
 ) !void {
-    const actual_request_json = try rpc.request.serialize(std.testing.allocator, request);
+    const request: rpc.request.Request = .{
```

Comment:
Originally these tests were covering the same code that's run by the client. But now they are not. You're serializing `Request` whereas the client serializes the inner request struct independently.